dbt/include/redshift/macros/adapters/apply_grants.sql

```diff
@@ -24,4 +62,72 @@ where has_table_privilege(u.usename, '{{ relation }}', privilege_type)
     and u.usename != current_user
     and not u.usesuper
 
+union all
+-- check that group has table privilege
+select
+    'group' as grantee_type,
+    g.groname as grantee,
+    p.privilege_type
+from pg_group g
+cross join privileges p
+where exists(
+    select *
+    from information_schema.table_privileges tp
+    where tp.grantee=g.groname
+    and tp.table_schema=replace(split_part('{{ relation }}', '.', 2), '"', '')
+    and tp.table_name=replace(split_part('{{ relation }}', '.', 3), '"', '')
+    and LOWER(tp.privilege_type)=p.privilege_type
+)
+
+union all
+-- check that role has table privilege
+select
+    'role' as grantee_type,
+    r.role_name as grantee,
+    p.privilege_type
+from svv_roles r
```

Comment:
From pairing with @dbeatty10 - in order to query `svv_table_info`, this requires:
- that dbt's user has role `<somerole>`
- `grant access system table to role <somerole>;`

References:
- https://docs.aws.amazon.com/redshift/latest/dg/r_SVV_ROLES.html
- https://www.redshiftresearchproject.org/white_papers/downloads/users_groups_roles_and_privileges.html